client/src/modules/stock/entry/modals/lots.modal.js

```diff
@@ -9,196 +9,142 @@ StockDefineLotsModalController.$inject = [
-    if(vm.entryType === 'transfer_reception') {
-      handleChange(vm.gridOptions.data[0]);
-    }
-
-  }
-
-  // remove lot
-  function removeLot(index) {
-    vm.gridOptions.data.splice(index, 1);
+    checkAll();
   }
 
-  // handle date change
-  function onDateChangeCallback(date, inventory) {
-    inventory.expiration_date = date;
-    // notify date change
-    handleChange(inventory);
-  }
-
-
-  function hasInvalidInventory(hasQuantity, hasExpiration) {
-    var error;
-
-    var isExpiration = hasExpiration ? false : true;
-    var isExcessiveQuatity = hasQuantity ? false : true;
-
-    if (isExcessiveQuatity) {
-      vm.errorText = 'STOCK.ERRORS.EXCESSIVE_QUANTITY';
-      error = true;
-    } else if (isExpiration) {
-      vm.errorText = 'STOCK.ERRORS.PLEASE_CHECK_EXPIRY_DATE';
-      error = true;
-    }
-
-    return error;
+  function onRegisterApi(api) {
+    vm.gridApi = api;
   }
 
-  // handleChange
-  function handleChange(inventory) {
-    vm.error =  false;
-
-    var sum = vm.gridOptions.data.reduce(sumQuantity, 0);  
-    var hasQuantity = (vm.inventory.quantity >= sum);
-    var hasLotLabel = inventory.lot;
-    var hasExpiration = (new Date(inventory.expiration_date) >= new Date());
-    inventory.is_valid = (hasQuantity && hasLotLabel && hasExpiration);
-
-    vm.error = hasInvalidInventory(hasQuantity, hasExpiration);
-    vm.submitError = vm.error ? vm.error : false;
+  function checkAll() {
+    vm.isLotExist = vm.stockLine.lots.length > 0;
 
+    vm.isEveryLineCorrect = vm.stockLine.lots.every(function (lot) {
+      return lot.isValid === true;
+    });
 
-    vm.remainingQuantity = (vm.inventory.quantity - sum >= 0) ? vm.inventory.quantity - sum : 0;
-    vm.sum = sum;
+    vm.sum = vm.stockLine.lots.reduce(function (x, y) {
+      return x + y.quantity;
+    }, 0);
 
-    if (vm.remainingQuantity) {
-      vm.maxLotReached = false;
-    }
-  }
+    vm.isQuantityCorrect = vm.sum <= vm.stockLine.quantity;
 
-  // check rows validity
-  function validLots() {
-    return vm.gridOptions.data.every(function (item) {
-      return item.is_valid === true;
-    });
+    vm.isAllValid = vm.isEveryLineCorrect && vm.isQuantityCorrect && vm.isLotExist;
   }
 
-  // submit
-  function submit(detailsForm) {
-    // This structure check if there are empty field
-    if (detailsForm.$invalid) {
-      vm.submitError = true;
-      vm.errorText = vm.error ? vm.error : 'FORM.ERRORS.RECORD_ERROR'; 
-
-      return;
-    } else {
-      vm.submitError = false;
-      // If the form is still at this level the only remaining reason is the expiration date
-
-      if (!validLots()) { 
-        vm.submitError = vm.error ? vm.error : false;
+  function checkLine(line, date) {
+    if (date) { line.expiration_date = date; }
 
-        return;         
-      }
-    }
+    var isPosterior = new Date(line.expiration_date).getTime() >= new Date().getTime();
```

Comment:
I think the `getTime()` call here is unnecessary.  It should work properly with just `new Date() > new Date()`.